src/PhpSpec/CodeGenerator/Generator/PromptingGenerator.php

```diff
@@ -39,20 +41,29 @@
      * @var \PhpSpec\Util\Filesystem
      */
     private $filesystem;
+
     /**
      * @var ExecutionContextInterface
      */
     private $executionContext;
 
     /**
-     * @param IO               $io
+     * @var EventDispatcherInterface
+     */
+    private $dispatcher;
+
+    /**
+     * @param IO $io
      * @param TemplateRenderer $templates
-     * @param Filesystem       $filesystem
+     * @param EventDispatcherInterface $dispatcher
+     * @param Filesystem $filesystem
+     * @param ExecutionContextInterface $executionContext
      */
-    public function __construct(IO $io, TemplateRenderer $templates, Filesystem $filesystem = null, ExecutionContextInterface $executionContext = null)
+    public function __construct(IO $io, TemplateRenderer $templates, EventDispatcherInterface $dispatcher, Filesystem $filesystem = null, ExecutionContextInterface $executionContext = null)
```

Comment:
To avoid BC breakage I'm not happy with changing the method signature in an incompatible way.

In other cases we've added optional arguments to the end of the dependency list, and you'd have to handle the case where an extension doesn't pass the dispatcher (by not dispatching that event).